lib/shopify_cli/theme/extension/dev_server/proxy_helpers/template_param_builder.rb

```diff
@@ -12,7 +12,7 @@ def build
               # Core doesn't support replace_extension_templates
```

Comment:
These changes make the `param_builder` classes very similar to each other. I'm wondering if we can create a `base_param_builder` and just pass in lambdas for the `build` and `as_param` methods?